build.gradle

```diff
@@ -60,6 +83,11 @@ scalastyle {
     }
 }
 
+task uiTest(type: Test) {
+    testClassesDirs = sourceSets.e2e.output.classesDirs
+    classpath = sourceSets.e2e.runtimeClasspath
+}
```

Comment:
No other task currently depends on this. Trigger it manually after you have run **gradle clean** and **gradle runIdeForUiTests** is running.

Reply:
can our custom **task** incorporate **gradle clean & gradle runIdeForUiTests**?

Reply:
Not easily, I'm afraid. The issue is, **runIdeForUiTests** should be run in parallel with **uiTest** and **clean** before **runIdeForUiTests**. AFAIK, Gradle doesn't support parallel tasks. The way they suggest running it [here](https://github.com/JetBrains/intellij-ui-test-robot#quick-start) is launching each task separately.

Reply:
yes, despite doing that myself I've missed the async part, I'm curious to see how it behaves in terms of `runIdeFor...` termination. guess I want that to-be-created "ui-tests-in-CI" task for myself 😉

Reply:
this one https://github.com/Aalto-LeTech/intellij-plugin/issues/517 @OlliKiljunen